internal/storage/storageutil/folder_attrs.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
```

Comment:
I think these are part of folder domain only, we can keep single file for folder domain, can we move these methods to folder.go file itself ? any concerns with that ?

Reply:
Isn't it more of helper functions only which should not be part of actual folder file? Also with that we will have to add controlpb dependency over there. Which we did not want to do I guess.

Reply:
As per Domain Driven Design , If helper functions change domain state, it should be part of domain [ref](https://medium.com/@chikim79/enriching-entity-classes-with-business-logic-with-domain-driven-design-principles-0a937983f529#:~:text=Benefits%20of%20Implementing%20Business%20Logic%20within%20the%20Entities)

Reply:
the purpose of restricting controlpb is not leak into aother business logics like other bucket wrappers and dir.go, folder domain is like an ACL to achieve that and is tighly coupled with control pb folder hence adding that dependency should not be an issue.

Reply:
got it done.